### ISSUE TITLE
Spam filtering model

### ISSUE THREAD

**Issue by rlucatoor** [tanaos/artifex#21]
## What is a Spam Filtering model

A Spam Filtering model predicts whether or not a text is spam.

## What needs to be done

Implement a Spam Filtering model that exposes the following public methods:

### `.train()` method:

```python
.train(output_path: Optional[str] = None, num_samples: int = config.DEFAULT_SYNTHEX_DATAPOINT_NUM, num_epochs: int = 3)`
```

The method should train the model to assign one of 2 labels — `spam` or `ham` — to the provided text. The `BaseModel._generate_synthetic_data()` method should be used to generate synthetic training data.

### `.__call__(text: Union[str, list[str])`

- If a `str` is provided, the model should return either `"spam"` or `"ham"`.

- If a `list[str]` is provided, the model should return a list of `"spam"` or `"ham"` strings, one per sentence.

### `.load(model_path: str)`

Loads a pre-trained Spam Filtering model from the specified path.
